hathor/utils/arg_parse.py

```diff
@@ -0,0 +1,24 @@
+#  Copyright 2023 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from argparse import Namespace
+from typing import NamedTuple, Type, TypeVar
+
+from hathor.utils.named_tuple import validated_named_tuple_from_dict
+
+T = TypeVar('T', bound=NamedTuple)
+
+
+def from_namespace(named_tuple_type: Type[T], namespace: Namespace) -> T:
+    return validated_named_tuple_from_dict(named_tuple_type, vars(namespace))
```

Comment:
I'm assuming `validated_named_tuple_from_dict` will ignore extra vars. Could you add a comment to highlight that it will silently ignore extra vars/parameters, so if you add arguments to `namespace` they will be ignored unless explicitly added to `RunNodeArgs`. I agree it's the best behavior though, because many subcommands have extra arguments that only make sense to be ignored here.

Reply:
You're right, sub commands that inherit from `RunNode` are not working. I'll implement tests to assert this, and then change the code to support them.

Also, I just noticed that the error I sent above is not from pydantic, but from `NamedTuple`. So I'll have to find another way to fix it.

Reply:
In https://github.com/HathorNetwork/hathor-core/pull/636/commits/483db339af90daafb6a8202d8274138b34449e59 I changed the implementation to use pydantic's `BaseModel` directly, instead of using a `NamedTuple` and then validating it with pydantic. By doing this, we have more flexibility.

The problem is fixed, and I've added tests to assert them